libsigner/src/events.rs

```diff
@@ -50,11 +50,22 @@ use wsts::state_machine::signer;
 use crate::http::{decode_http_body, decode_http_request};
 use crate::{EventError, SignerMessage};
 
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+/// BlockProposal sent to signers
+pub struct BlockProposalSigners {
+    /// The block itself
+    pub block: NakamotoBlock,
+    /// The burn height the block is mined during
+    pub burn_height: u64,
+    /// The reward cycle the block is mined during
+    pub reward_cycle: u64,
```

Comment:
I don't think this field is necessary, since it's a derived value from the `burn_height`.  I don't think it should be possible to ever construct one of these structs with inconsistent `burn_height` and `reward_cycle` values.

Reply:
The signer does not currently have the information needed to map burn height to reward cycle, so it would otherwise require a call to the stacks-node to compute.

Reply:
This is fine for now, but PoX constants are well-known and unchanging so it's safe to rely on them in the signer (no RPC required).